src/connector/src/source/pulsar/mod.rs

```diff
@@ -43,4 +60,36 @@ pub struct PulsarProperties {
 
     #[serde(rename = "auth.token")]
     pub auth_token: Option<String>,
+
+    #[serde(flatten)]
+    pub oauth: Option<PulsarOauth>,
+}
+
+impl PulsarProperties {
+    pub async fn build_pulsar_client(&self) -> Result<Pulsar<TokioExecutor>> {
+        let mut pulsar_builder = Pulsar::builder(&self.service_url, TokioExecutor);
+        if let Some(oauth) = &self.oauth {
+            let url = Url::parse(&oauth.credentials_url)?;
+            if url.scheme() == "s3" {
+                todo!("s3 oauth credentials not supported yet");
+            }
+
+            let auth_params = OAuth2Params {
+                issuer_url: oauth.issuer_url.clone(),
+                credentials_url: oauth.credentials_url.clone(),
+                audience: Some(oauth.audience.clone()),
+                scope: oauth.scope.clone(),
+            };
+
+            pulsar_builder = pulsar_builder
+                .with_auth_provider(OAuth2Authentication::client_credentials(auth_params));
+        } else if let Some(auth_token) = &self.auth_token {
```

Comment:
It seems `auth_token` and `oauth` are mutually exclusive?

Do we need to inform users that although both are set, only `auth_token` will be considered?

Reply:
pulsar supports these two auth methods. I think in most cases, user won't provide both fields. and of course, I will mention it in doc